### PR TITLE
Reduce to single blockstore inserter thread in BroadcastStage

### PR DESCRIPTION
**Pull request by steviez**
#### Problem
The implementation of Blockstore::insert_shreds() holds a write lock for the entirety of the function. Thus, there is no point in having more than one thread in BroadcastStage insert shreds. Namely, this lock:
https://github.com/anza-xyz/agave/blob/1eaa5cf1aa8a8713db9ea76c4c99fb37c71cd89c/ledger/src/blockstore.rs#L941

#### Summary of Changes
Reduce the constant to have **one** thread instead of **two**

Kudos to the Firedancer team (not sure who exactly first observed this) for bringing this to our attention

### CODE REVIEW

**Review by bw-solana — DISMISSED** [review]
LGTM

File: turbine/src/broadcast_stage.rs [hunk 1]
```diff
@@ -56,7 +56,9 @@ mod standard_broadcast_run;
 const CLUSTER_NODES_CACHE_NUM_EPOCH_CAP: usize = 8;
 const CLUSTER_NODES_CACHE_TTL: Duration = Duration::from_secs(5);
 
-pub(crate) const NUM_INSERT_THREADS: usize = 2;
+// Blockstore::insert_threads() obtains and holds a write lock for the entire function. Until
+// this changes, there is no point in having more than one blockstore inserter thread.
+pub(crate) const NUM_INSERT_THREADS: usize = 1;
```
Comment (behzadnouri) [inline turbine/src/broadcast_stage.rs]: Should we remove this constant entirely and simplify:
https://github.com/anza-xyz/agave/blob/cde467dc8/turbine/src/broadcast_stage.rs#L334-L352

**Review by behzadnouri — DISMISSED** [review]
LGTM

File: turbine/src/broadcast_stage.rs (hunk 1 above)
Comment (steviez) [reply turbine/src/broadcast_stage.rs]: I could go either way. Hypothetically, if we remove the lock in Blockstore, more threads could potentially be beneficial. But, given that Blockstore isn't the bottleneck, I'm not sure we'll spend much time on it for a while so the simplification you suggested could be good

File: turbine/src/broadcast_stage.rs [hunk 2]
```diff
@@ -331,25 +329,26 @@ impl BroadcastStage {
...
-            })
-            .take(NUM_INSERT_THREADS),
-        );
+
+        // Blockstore::insert_threads() obtains and holds a write lock for the entire function.
+        // Until this changes, only a single inserter thread is necessary.
+        thread_hdls.push({
+            let blockstore_receiver = blockstore_receiver.clone();
+            let mut bs_record = broadcast_stage_run.clone();
+            let btree = blockstore.clone();
```
Comment (behzadnouri) [inline turbine/src/broadcast_stage.rs]: some of these `.clone`s are probably redundant now.

**Review by behzadnouri — DISMISSED** [review]
lgtm

File: turbine/src/broadcast_stage.rs (hunk 2 above)
Comment (steviez) [reply turbine/src/broadcast_stage.rs]: Good call, I saw the `move || loop` and ignored thinking about this but we can do away with 2 of them. Decided to fix it now

**Review by behzadnouri — APPROVED** [review]

**Pull request merged**